pennylane/pulse/parametrized_evolution.py

```diff
@@ -172,30 +172,33 @@ def __init__(
             raise ValueError(
                 "All operators inside the parametrized hamiltonian must have a matrix defined."
             )
+        self._has_matrix = params is not None and t is not None
         self.H = H
-        self.params = params
         self.odeint_kwargs = odeint_kwargs
         if t is None:
             self.t = None
         else:
             self.t = jnp.array([0, t] if qml.math.ndim(t) == 0 else t, dtype=float)
-        super().__init__(wires=H.wires, do_queue=do_queue, id=id)
+        params = [] if params is None else params
+        super().__init__(*params, wires=H.wires, do_queue=do_queue, id=id)
 
     def __call__(self, params, t, **odeint_kwargs):
-        self.params = params
+        self.data = [jnp.array(p) if isinstance(p, list) else p for p in params]
+        self._num_params = len(params)
```

Comment:
Here we basically mimic parts of what is going on in `Operation.__init__`, right?
How fragile would you say is this construction? Should we leave a comment regarding this?

Reply:
I changed this method. I don't like updating classes dynamically, thus now `__call__` returns a **new** operator.

Reply:
I like this.